examples/simple-genai-server/main.go

```diff
@@ -0,0 +1,240 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"agones.dev/agones/pkg/util/signals"
+	sdk "agones.dev/agones/sdks/go"
+)
+
+// Main starts a server that serves as an example of how to integrate GenAI endpoints into your dedicated game server.
+func main() {
+	sigCtx, _ := signals.NewSigKillContext()
+
+	port := flag.String("port", "7654", "The port to listen to traffic on")
+	genAiEndpoint := flag.String("GenAiEndpoint", "", "The full base URL to send API requests to to simulate computer (NPC) responses to user input")
+	genAiContext := flag.String("GenAiContext", "", "Context for the GenAI endpoint")
+	prompt := flag.String("Prompt", "", "The first prompt for the GenAI endpoint")
+	simEndpoint := flag.String("SimEndpoint", "", "The full base URL to send API requests to to simulate user input")
+	simContext := flag.String("SimContext", "", "Context for the Sim endpoint")
+	numChats := flag.Int("NumChats", 1, "Number of back and forth chats between the sim and genAI")
+
+	flag.Parse()
+	if ep := os.Getenv("PORT"); ep != "" {
+		port = &ep
+	}
+	if sc := os.Getenv("SimContext"); sc != "" {
+		simContext = &sc
+	}
+	if gac := os.Getenv("GenAiContext"); gac != "" {
+		genAiContext = &gac
+	}
+	if p := os.Getenv("Prompt"); p != "" {
+		prompt = &p
+	}
+	if se := os.Getenv("SimEndpoint"); se != "" {
+		simEndpoint = &se
+	}
+	if gae := os.Getenv("GenAiEndpoint"); gae != "" {
+		genAiEndpoint = &gae
+	}
+	if nc := os.Getenv("NumChats"); nc != "" {
+		num, err := strconv.Atoi(nc)
+		if err != nil {
+			log.Fatalf("Could not parse NumChats: %v", err)
+		}
+		numChats = &num
+	}
+
+	log.Print("Creating SDK instance")
+	s, err := sdk.NewSDK()
+	if err != nil {
+		log.Fatalf("Could not connect to sdk: %v", err)
+	}
+
+	log.Print("Starting Health Ping")
+	go doHealth(s, sigCtx)
+
+	var simConn *connection
+	if *simEndpoint != "" {
+		log.Printf("Creating Sim Client at endpoint %s", *simEndpoint)
+		simConn = initClient(*simEndpoint, *simContext, "Sim")
+	}
+
+	if *genAiEndpoint == "" {
+		log.Fatalf("GenAiEndpoint must be specified")
+	}
+	log.Printf("Creating GenAI Client at endpoint %s", *genAiEndpoint)
+	genAiConn := initClient(*genAiEndpoint, *genAiContext, "GenAI")
+
+	log.Print("Marking this server as ready")
+	if err := s.Ready(); err != nil {
+		log.Fatalf("Could not send ready message")
+	}
+
+	// Start up TCP listener so the user can interact with the GenAI endpoint manually
+	if simConn == nil {
+		go tcpListener(*port, genAiConn)
+	} else {
+		log.Printf("Starting autonomous chat with Prompt: %s", *prompt)
+		go autonomousChat(*prompt, genAiConn, simConn, *numChats)
```

Comment:
Do you think that is autonomous mode the gameserver should exit once the conversation is done (maybe with an optional sleep time in case you want to grab the logs)? When running it I find it a bit strange that the gameserver just keeps running in the ready state once it's done working but continues to send health pings.